todo/android/todo/src/main/java/com/mongodb/stitch/sdk/examples/todo/MainActivity.java

```diff
@@ -98,34 +99,13 @@ public void onLogin() {
         }
 
         @Override
-        public void onLogout(final String lastProvider) {
+        public void onLogout() {
             final MainActivity activity = _main.get();
 
             final List<Task<Void>> futures = new ArrayList<>();
             if (activity != null) {
                 activity._handler.removeCallbacks(activity._refresher);
 
-                switch (lastProvider) {
-                    case GoogleAuthProviderInfo.FQ_NAME:
-                        if (activity._googleApiClient != null && activity._googleApiClient.isConnected()) {
-                            final TaskCompletionSource<Void> future = new TaskCompletionSource<>();
-                            com.google.android.gms.auth.api.Auth.GoogleSignInApi.signOut(
-                                    activity._googleApiClient).setResultCallback(new ResultCallback<Status>() {
-                                @Override
-                                public void onResult(@NonNull final Status ignored) {
-                                    future.setResult(null);
-                                }
-                            });
-                            futures.add(future.getTask());
-                        }
-                        break;
-                    case FacebookAuthProviderInfo.FQ_NAME:
-                        if (activity._fbInitOnce) {
-                            LoginManager.getInstance().logOut();
-                        }
-                        break;
-                }
-
```

Comment:
This entire `switch` seems unnecessary now (everything seems to work without it....but we don't have a test suite...). Would be great if someone could verify. I removed it because the new onLogout() doesn't take any parameters, so lastProvider isn't directly available to us.

Reply:
The main point of this switch was to log people out of the Facebook and/or Google SDK if that's what they used to sign into Stitch. If there's a way in the Facebook and Google SDKs to check whether or not a user is currently signed in with their respective service, we should use that and log people out of whatever they're logged into to properly clean up.